### PR TITLE
PHPCS cleanup and PHP cs fixer config

### LINKED ISSUE #198: Code cleanup pass
Look over all the code, check for:

- Unreachable code (e.g. functions that are never called).
- Files without trailing new lines.
- Tabs that should be spaces.
- Consistent capitalization (e.g. some static methods are CamelCase some are whateverThisIsCalled)
- Comments are correct (and all the methods that need documentation have doc comments).

### PR DESCRIPTION
**Pull request by larowlan**
Fixes #198 
Branched from #197 

Changes as follows:
* Uses https://github.com/FriendsOfPHP/PHP-CS-Fixer to automate as much as possible, including adding a config file. For future cleanups, just run `php-cs-fixer fix` from project root after installing globally.
* Adds missing doc-blocks, fixes some class not found issues and various other issues reported by php-storm inspections

### CODE REVIEW

**Comment by defuse** [comment]
Thanks!

-1 to including the license text in the files. It's distracting and apparently for legal reasons it's super difficult to ever remove it in the future.

**Comment by defuse** [comment]
Can we get the same changes but without the license in the files? :)

**Comment by larowlan** [comment]
Sure, will update .php_cs to remove it too

**Comment by larowlan** [comment]
Ok, removed licenses and updated php cs fixer config

File: src/File.php [hunk 1]
```diff
@@ -372,281 +379,281 @@ public static function decryptResource($inputHandle, $outputHandle, Key $key)
...
+         *
+         * $ekey -- Encryption Key -- used for AES
+         */
+        $ekey = Core::HKDF(
+            Core::HASH_FUNCTION_NAME,
+            $key->getRawBytes(),
+            Core::KEY_BYTE_SIZE,
+            Core::ENCRYPTION_INFO_STRING,
+            $file_salt
+        );
```
Comment (defuse) [inline src/File.php]: Oh god, the diff is going all screwy here. I *think* it's just whitespace changes but Github isn't highlighting them, so I can't tell if it's just whitespace or if this commit inserts hidden backdoors!

Comment (larowlan) [inline src/File.php]: Does https://github.com/defuse/php-encryption/pull/200/files?w=1 help - the ?w=1 should ignore whitespace changes

**Comment by defuse** [comment]
LGTM. I was careful to review it for subtly-inserted backdoors and I couldn't find any. Just to be *super* cautious I'm going to redo the File.php changes myself as I'm merging, since GitHub's diff is glitching and I can't easily see that it's only whitespace changes.

(Underhanded Crypto contest entry idea: Find a bug in GitHub's diff algorithm to make it look like whitespace changes when it's not).

File: src/File.php [hunk 2]
```diff
@@ -372,281 +379,281 @@ public static function decryptResource($inputHandle, $outputHandle, Key $key)
...
+        /**
+         * Reset file pointer to the beginning of the file after the header
+         */
+        if (\fseek($inputHandle, Core::HEADER_VERSION_SIZE, SEEK_SET) === false) {
+            throw new Ex\CannotPerformOperationException(
+                'Cannot read seek within input file'
+            );
+        }
 
-            $thisIv = $iv;
```
Comment (defuse) [inline src/File.php]: It's scary how easy a backdoor could be hidden in all of this!

File: src/File.php (hunk 1 above)
Comment (defuse) [inline src/File.php]: @larowlan: Oh yes, it very much does help, thanks!

Comment (defuse) [inline src/File.php]: Actually it still screws up displaying the decryptResource stuff :( Oh well

**Comment by defuse** [comment]
I just merged @larowlan's PR without applying the changes to `File.php` myself. It was too much work to do it manually. Instead, I looked over the (buggy) diff for a couple minutes and tried to make sure the code was the same. I think it is, but I can't be certain. Hopefully @larowlan didn't just insert a backdoor into our library!

@larowlan: Thanks so much for this! The code looks a lot better now :)

File: src/File.php (hunk 2 above)
Comment (defuse) [inline src/File.php]: Note to others reading: This isn't actually a backdoor, the line isn't *actually* removed, the diff is just glitching.

**Comment by paragonie-scott** [comment]
I looked over it last night. Per #110 we're going to audit this again before v2.0.0 is tagged, so we'll find out either way.

**Comment by larowlan** [comment]
Neat, thanks

File: src/File.php (hunk 1 above)
Comment (redragonx) [inline src/File.php]: Looks like it hates extra spaces.

**Comment by narfbg** [comment]
Structural, functional and styling changes don't belong together in the same patch - that's why it's so hard to review this.

**Comment by defuse** [comment]
@narfbg: Yeah, I understand. You can still review commit-by-commit.

**Comment by defuse** [comment]
@narfbg: Oh whoops, I thought we were on #212 nevermind!

**Comment by narfbg** [comment]
Well, even if that was the case, I'd still argue in favor of a policy to limit the scope of separate PRs. That's what I was trying to imply here. :)

**Pull request merged**